src/js/rtc_peer_connection_factory.js

```diff
@@ -112,7 +117,11 @@ export default class RtcPeerConnectionFactory {
     _refreshRtcPeerConnection() {
         this._clearIdleRtcPeerConnection();
         this._logger.log("refreshing peer connection for client " + this._clientId);
-        this._requestPeerConnection();
+        if (!this._peerConnectionRequestInFlight) {
+            this._peerConnectionRequestInFlight = true;
+            this._requestPeerConnection();
```

Comment:
Are we trying to achieve mutex here? Is there a possible of race condition in multi tab/browser setup?

Reply:
Yes. This code executes async, hence there is a possibility of race condition in multi tab scenario